src/functions.jl

```diff
@@ -107,6 +113,17 @@ if VERSION >= v"1.1"
     end
 end
 
+@static if VERSION > v"1.9-DEV"
+    function Base.stack(A::KeyedArray; dims::Colon=:)
+        data = @invoke Base.stack(A::AbstractArray; dims)
+        if !allequal(named_axiskeys(a) for a in A)
+            throw(DimensionMismatch("stack expects uniform axiskeys for all arrays"))
+        end
+        akeys = (; named_axiskeys(first(A))..., named_axiskeys(A)...)
```

Comment:
This won't be correct with `dims=1`.

Reply:
Oh now I see `dims::Colon=:`. Maybe at least that should come with a TODO comment?